language/tools/move-cli/src/package.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use include_dir::{include_dir, Dir, DirEntry};
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::Path,
+};
+use tempfile::tempdir;
+
+use move_lang::{compiled_unit::CompiledUnit, find_filenames, move_compile};
+use stdlib::{COMPILED_EXTENSION, MOVE_EXTENSION};
+use vm::file_format::CompiledModule;
+
+/// Directory name for the package source files under package/<name>
+const PKG_SOURCE_DIR: &str = "source_files";
+/// Directory name for the package binary files under package/<name>
+const PKG_BINARY_DIR: &str = "compiled";
+
+/// Content for the stdlib package
+const PACKAGE_STDLIB: MovePackage = MovePackage::pkg_stdlib();
+/// Content for the nursery package
+const PACKAGE_NURSERY: MovePackage = MovePackage::pkg_nursery();
+
+struct MovePackage {
+    /// Name of the package
+    name: &'static str,
+    /// The directory containing all the .move source files
+    source_dir: Dir<'static>,
+    /// The directory containing all the .mv binary files, if available
+    binary_dir: Option<Dir<'static>>,
+    /// Dependencies
+    deps: &'static [&'static MovePackage],
+}
+
+impl MovePackage {
+    /// Build the stdlib package statically
+    const fn pkg_stdlib() -> Self {
+        MovePackage {
+            name: "stdlib",
+            source_dir: include_dir!("../../stdlib/modules"),
+            binary_dir: Some(include_dir!("../../stdlib/compiled/stdlib")),
```

Comment:
Don't include this, we need to regenerate them

Reply:
You mean, compile them when, say, we first call `load_modules` on this package?

Reply:
I would compile them ahead of time to keep it simple, but yeah thats the idea

Reply:
You'll probably want to either compile them least every time the mtime changes or every time you call `prepare_source_files` if you don't want to deal with tracking mtimes.

Reply:
I think with `include_dir!`, we are statically include a snapshot of the file content with our CLI binary at build time, so I guess we do not need to worry about `mtimes`? Correct me if I misinterpret the message.

Reply:
I guess I was assuming if the files existed on disk, it was to support some workflow where you'd update them without having to recompile the CLI, but I suppose you'd just use `--mode bare` for that. If the files just exist because the compiler API can only take a `Vec<String>` as input, perhaps we should consider support `Vec<Box<dyn trait io::Read>>` to avoid round tripping to disk.